include/xtensor/xstridedview.hpp

```diff
@@ -0,0 +1,567 @@
+    inline auto xstrided_view<CT, CD>::stepper_end(const ST& shape) -> stepper
+    {
+        size_type offset = shape.size() - dimension();
+        return stepper(this, offset, true);
+    }
+
+    template <class CT, class CD>
+    template <class ST>
+    inline auto xstrided_view<CT, CD>::stepper_begin(const ST& shape) const -> const_stepper
+    {
+        size_type offset = shape.size() - dimension();
+        return const_stepper(this, offset);
+    }
+
+    template <class CT, class CD>
+    template <class ST>
+    inline auto xstrided_view<CT, CD>::stepper_end(const ST& shape) const -> const_stepper
+    {
+        size_type offset = shape.size() - dimension();
+        return const_stepper(this, offset, true);
+    }
+
+    /**
+     * Construct a strided view from an xexpression, shape, strides and offset.
+     *
+     * @param e xexpression
+     * @param shape the shape of the view
+     * @param strides the new strides of the view
+     * @param offset the offset of the first element in the underlying container
+     *
+     * @tparam E type of xexpression
+     * @tparam I shape and strides type
+     *
+     * @return the view
+     */
+    template <class E, class I>
+    inline auto strided_view(E&& e, I&& shape, I&& strides, std::size_t offset = 0) noexcept
+    {
+        using view_type = xstrided_view<xclosure_t<E>, decltype(e.data())>;
+        return view_type(std::forward<E>(e), std::forward<I>(shape), std::forward<I>(strides), offset);
+    }
+
+    /****************************
+     * transpose implementation *
+     ****************************/
+
+    namespace detail
+    {
+        template <class E, class S>
+        inline auto transpose_impl(E&& e, S&& permutation, check_policy::none)
+        {
+            if (container_size(permutation) != e.dimension())
+            {
+                throw transpose_error("Permutation does not have the same size as shape");
+            }
+
+            // permute stride and shape
+            using strides_type = typename std::decay_t<E>::strides_type;
+            strides_type temp_strides;
+            resize_container(temp_strides, e.strides().size());
+
+            using shape_type = typename std::decay_t<E>::shape_type;
+            shape_type temp_shape;
+            resize_container(temp_shape, e.shape().size());
+
+            // shape_type temp_backstrides;
+            // resize_container(temp_backstrides, backstrides().size());
+
+            for (std::size_t i = 0; i < e.shape().size(); ++i)
+            {
+                if (std::size_t(permutation[i]) >= e.dimension())
+                {
+                    throw transpose_error("Permutation contains wrong axis");
+                }
+                temp_shape[i] = e.shape()[permutation[i]];
+                temp_strides[i] = e.strides()[permutation[i]];
+                // TODO figure out what to do with backstrides
+                // temp_backstrides[i] = backstrides()[permutation[i]];
```

Comment:
I think you can use the `adapt_strides` method in `xstrides.hpp` to compute the backstrides.